projects/components/src/menu-dropdown/menu-dropdown.component.test.ts

```diff
@@ -2,14 +2,12 @@ import { fakeAsync } from '@angular/core/testing';
 import { IconType } from '@hypertrace/assets-library';
 import { MenuDropdownComponent, MenuItemComponent } from '@hypertrace/components';
 import { createHostFactory, SpectatorHost } from '@ngneat/spectator/jest';
-import { MockComponent } from 'ng-mocks';
 import { LabelComponent } from '../label/label.component';
 
 describe('Menu dropdown Component', () => {
   const hostFactory = createHostFactory<MenuDropdownComponent>({
     component: MenuDropdownComponent,
-    entryComponents: [MenuItemComponent],
-    declarations: [MockComponent(LabelComponent)],
+    declarations: [LabelComponent, MenuItemComponent],
```

Comment:
Using the real components here means this isn't a shallow test - if that's intentional, let's remove line 11. If not, let's mock the menu and label components.

Reply:
Will remove shallow then. For this test to work, we need the MenuItemComponent. Mocking doesn't work. Or we have to use entryComponent

Reply:
That's only because the test is written to assume the internals of the item. The following change removes that assumption and keeps things shallow.
```diff
diff --git a/projects/components/src/menu-dropdown/menu-dropdown.component.test.ts b/projects/components/src/menu-dropdown/menu-dropdown.component.test.ts
index 878d6e9..cabcd7d 100644
--- a/projects/components/src/menu-dropdown/menu-dropdown.component.test.ts
+++ b/projects/components/src/menu-dropdown/menu-dropdown.component.test.ts
@@ -8,7 +8,7 @@ import { LabelComponent } from '../label/label.component';
 describe('Menu dropdown Component', () => {
   const hostFactory = createHostFactory<MenuDropdownComponent>({
     component: MenuDropdownComponent,
-    declarations: [MockComponent(LabelComponent), MenuItemComponent],
+    declarations: [MockComponent(LabelComponent), MockComponent(MenuItemComponent)],
     shallow: true
   });
 
@@ -36,11 +36,11 @@ describe('Menu dropdown Component', () => {
     );
 
     spectator.click('.trigger-content');
-    const optionElements = spectator.queryAll('.menu-item', { root: true });
-    expect(spectator.query('.dropdown-content', { root: true })).toExist();
-    expect(optionElements.length).toBe(2);
+    const optionComponents = spectator.queryAll(MenuItemComponent);
+    expect(spectator.query('.dropdown-content')).toExist();
+    expect(optionComponents.length).toBe(2);
 
-    expect(optionElements[0]).toHaveText('Do X');
-    expect(optionElements[1]).toHaveText('Do Y');
+    expect(optionComponents[0].label).toBe('Do X');
+    expect(optionComponents[1].label).toBe('Do Y');
   }));
 });

```

Reply:
aah. interesting. thanks!